controllers/release/adapter.go

```diff
@@ -180,14 +182,58 @@ func (a *adapter) EnsureReleaseIsRunning() (controller.OperationResult, error) {
 	return controller.ContinueProcessing()
 }
 
-// EnsureReleaseIsProcessed is an operation that will ensure that a managed Release PipelineRun associated to the Release
+// EnsureTenantReleasePipelineIsProcessed is an operation that will ensure that a Tenant Release PipelineRun associated to the Release
 // being processed and a RoleBinding to grant its serviceAccount permissions exist. Otherwise, it will create them.
-func (a *adapter) EnsureReleaseIsProcessed() (controller.OperationResult, error) {
+func (a *adapter) EnsureTenantReleasePipelineIsProcessed() (controller.OperationResult, error) {
 	if a.release.HasProcessingFinished() {
 		return controller.ContinueProcessing()
 	}
 
-	pipelineRun, err := a.loader.GetManagedReleasePipelineRun(a.ctx, a.client, a.release)
+	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release)
+	if err != nil && !errors.IsNotFound(err) {
+		return controller.RequeueWithError(err)
+	}
+
+	if pipelineRun == nil || !a.release.IsProcessing() {
+		releasePlan, err := a.loader.GetReleasePlan(a.ctx, a.client, a.release)
+		if err != nil {
+			return controller.RequeueWithError(err)
+		}
+
+		if releasePlan.Spec.Pipeline == nil {
+			// no tenant pipeline to run
+			return controller.ContinueProcessing()
+		}
+
+		if pipelineRun == nil {
+			snapshot, err := a.loader.GetSnapshot(a.ctx, a.client, a.release)
+			if err != nil {
+				return controller.RequeueWithError(err)
+			}
+
+			pipelineRun, err = a.createTenantPipelineRun(releasePlan, snapshot)
+			if err != nil {
+				return controller.RequeueWithError(err)
+			}
+
+			a.logger.Info(fmt.Sprintf("Created %s Release PipelineRun", metadata.TenantPipelineType),
+				"PipelineRun.Name", pipelineRun.Name, "PipelineRun.Namespace", pipelineRun.Namespace)
+		}
+
+		return controller.RequeueOnErrorOrContinue(a.registerProcessingData(pipelineRun, nil))
+	}
+
+	return controller.ContinueProcessing()
+}
+
+// EnsureManagedReleasePipelineIsProcessed is an operation that will ensure that a managed Release PipelineRun associated to the Release
+// being processed and a RoleBinding to grant its serviceAccount permissions exist. Otherwise, it will create them.
+func (a *adapter) EnsureManagedReleasePipelineIsProcessed() (controller.OperationResult, error) {
+	if a.release.HasProcessingFinished() {
+		return controller.ContinueProcessing()
+	}
+
+	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release)
```

Comment:
I did not noticed any misbehavior while testing this. I will look deeper it the coding is actually stopping further processing after the tenant pipeline finishes and it gets to this point.

Reply:
A tenant pipeline can't be found in this function because this is the managed operation and we don't let a user define both a tenant and managed pipelinerun

Reply:
This is working as expected. Something else it not adding the `Released`  condition to it.

Reply:
Ok, so it works because only one kind of Pipeline is allowed. I don't like this way of coding. Not resilient enough. I'm ok to let it go through for the matter of merging this PR as soon as we can, but that needs a refactor IMHO.

Reply:
This way of coding like in assuming that something else prevented the behavior. Operations depending on other operations behavior. Can be fixed later on.